README.md
# InteractúaMóvil SDK PHP

SDK PHP para el API SMS de InteractúaMóvil.

Es necesario poseer un **API_KEY**, un **API_SECRET_KEY** y **API_URL**
para utilizar el API.

## Requerimientos
* PHP >= 5.1.2
* PECL hash >= 1.1

Ejemplo de creación de instancia del api:
    
```php
    require 'im-contactosms-sdk-php/src/SmsApi.php';

    
    /* Quiero mis respuestas como StdObj... querías arrays? 
        cambia el último parámetro a true */ 
    $api = new SmsApi(API_KEY, API_SECRET, API_URL, false);  

```

Para hacer llamadas al API pueden utilizarse los métodos `contacts()`, `groups()` y `messages()`  en el objeto de api, el formato de la respuesta es:

```php

    (array/StdObj) response
    [
        status => (string)"OK",
        ok => (boolean)true,
        code => (int)200,
        headers => (array)[/* the headers */]
        data => (array/StdObj)[/*the data*/]
    ];

    // ejemplos
    
    //    object(stdClass) (5) {
    //       ["code"]=>
    //       int(500)
    //       ["status"]=>
    //       string(5) "Error"
    //       ["ok"]=>
    //       bool(false)
    //       ["response_headers"]=>
    //       array(4) {
    //         [0]=>
    //         string(34) "HTTP/1.1 500 Internal Server Error"
    //         [1]=>
    //         string(30) "Content-Type: application/json"
    //         [2]=>
    //         string(42) "Server: ContactoSMS API/3.0 (Simple 5.1.4)"
    //         [3]=>
    //         string(17) "Connection: close"
    //       }
    //       ["data"]=>
    //       object(stdClass)#5 (2) {
    //         ["code"]=>
    //         int(503)
    //         ["error"]=>
    //         string(25) "Error interno. Reintentar"
    //       }
    //     }

    // array(5) {
    //   ["code"]=>
    //   int(200)
    //   ["status"]=>
    //   string(2) "OK"
    //   ["ok"]=>
    //   bool(true)
    //   ["response_headers"]=>
    //   array(4) {
    //     [0]=>
    //     string(15) "HTTP/1.1 200 OK"
    //     [1]=>
    //     string(30) "Content-Type: application/json"
    //     [2]=>
    //     string(42) "Server: ContactoSMS API/3.0 (Simple 5.1.4)"
    //     [3]=>
    //     string(17) "Connection: close"
    //   }
    //   ["data"]=>
    //   array(1) {
    //     [0]=>
    //     array(7) {
    //       ["msisdn"]=>
    //       string(11) "50212345678"
    //       ["status"]=>
    //       string(9) "SUSCRIBED"
    //       ["phone_number"]=>
    //       string(8) "12345678"
    //       ["country_code"]=>
    //       string(3) "502"
    //       ["first_name"]=>
    //       string(7) "Alberto"
    //       ["full_name"]=>
    //       string(7) "Alberto"
    //       ["added_from"]=>
    //       string(8) "WEB_FORM"
    //     }
    //   }
    // }



```
Ejemplos de llamadas al API:

```php

    $contacts = $api->contacts()
        ->getContacts('12345678' /* Or a name, it works too */,
        /*limit*/ 10,/* offset */ 0, /* contact status */'SUSCRIBED');

    if ($response->ok){
        /* Do something */
        echo "Mis contactos son: \n";
        foreach ($response->data as $contact){
            echo "$contact->msisdn : $contact->first_name\n";
        }
    }


    $groups = $api->groups()
        ->getGroups("my group" /*$query*/, 0 /*$limit*/,0 /* $offset */,
         false /*$shortResults*/);

    if ($groups->ok){
        echo "Mis grupos\n";
        foreach ($groups->data as $group){
            echo " Grupo: $group->name, miembros: {$group->members->total}\n";
        }
    }

    
    $message = $api->messages()
        ->sendToContact("50212345678", "Sent from PHP SDK");

    if ($message->ok) echo "Mensaje enviado..."

```

## Ejemplos

Puedes ver ejemplos de todas las funciones en [examples](https://github.com/interactuamovil/im-contactosms-sdk-php/tree/master/examples).


## Documentación

La documentación del SDK PHP se encontra en el wiki: [Ver documentación](https://github.com/interactuamovil/im-contactosms-sdk-php/wiki)

changelog.md
# Changelog

## v3.0:
    * Release para la versión 3 de ContactoSMS.
    * Ahora el SDK es dirigida a la versión 3.

examples/contacts.php
<?php

include("../src/SmsApi.php");

define('API_KEY', 'Your api key');
define('API_SECRET', 'Your api secret');
define('API_URL', 'Your api url');

$api = new SmsApi(API_KEY, API_SECRET, API_URL, true); // hey you'll have your responses in arrays, did you want StdObject? change last parameter to false

// Response format:
// $response = [
//     'status' => 'OK',
//     'ok' => true,
//     'code' => '200',
//     'headers' => [/* the headers */]
//     'data' => [/*the data*/]
// ];



/// Creating a contact......
echo ("Creating contact...");
$contact = $api->contacts()->createContact("12345678","502" /*, $firstName, $lastName, 
        $custom_field1,$custom_field2, $custom_field3, $custom_field4, $custom_field5*/);
if ($response['ok']) echo "I successfully created a contact!\n";

/// Retrieving a contact...

echo ("Getting contacts...\n");
// $api->contacts()->getContacts($query=null, $limit=null,$start=null,$status=null,$shortResults=null);
$contacts = $api->contacts()->getContacts('12345678' /* Or a name, it works too */,10,0,'SUSCRIBED');

// check if my answer was OK
if ($contacts['ok']){ /* alternative: $contacts['code']==200 */
    echo "All my contacts are: \n";
    foreach ($contacts['data'] as $contact){
        echo "Phone Number: {$contact['phone_number']}, Name: {$contact['full_name']}\n";
    }
} 

/// Updating........
echo ("Updating contact...");
// $api->conacts()->updateContact($msisdn, $phoneNumber=null, $countryCode=null, $firstName=null, $lastName = null,$custom_field1=null,$custom_field2=null, $custom_field3=null, $custom_field4=null, $custom_field5=null);
$response = $api->contacts()->updateContact("50212345678","12345678","502","Alberto");
if ($response['ok']) echo "Updated my contact\n";

/// Getting groups
echo ("Getting contact groups...\n");
$response = $api->contacts()->getContactGroups("50212345678");
echo "My contact belongs to:\n";
if ($response['ok']) foreach ($response['data'] as $group){
    echo $group['name']."\n";
}

/// Deleting a contact....
echo ("Deleting contact...");
$response = $api->contacts()->deleteContact("50212345678");
if ($response['ok']) echo "Contact deleted\n";
else {
    echo "Something went wrong here is the data";
    var_dump($response);
}
